src/generator/AutoRest.Go/Model/ParameterGo.cs

```diff
@@ -0,0 +1,238 @@
+        }
+
+        /// <summary>
+        /// Return formatted value string for the parameter.
+        /// </summary>
+        /// <returns></returns>
+        public string ValueForMap()
+        {
+            if (SerializedName.Value.IsApiVersion())
+            {
+                return "client." + ApiVersionName;
+            }
+            var value = IsClientProperty
+                ? "client." + CodeNamerGo.Instance.PascalCase(Name.Value)
+                : Name.Value;
+
+            var format = IsRequired || ModelType.CanBeEmpty()
+                                          ? "{0}"
+                                          : "*{0}";
+
+            var s = CollectionFormat != CollectionFormat.None
+                                  ? $"{format},\"{CollectionFormat.GetSeparator()}\""
+                                  : $"{format}";
+
+            return string.Format(
+                RequiresUrlEncoding()
+                    ? $"autorest.Encode(\"{Location.ToString().ToLower()}\",{s})"
+                    : $"{s}",
+                value);
+        }
+    }
+
+    public static class ParameterGoExtensions
+    {
+        /// <summary>
+        /// Return a Go map of required parameters.
+        // Refactor -> Generator
+        /// </summary>
+        /// <param name="parameters"></param>
+        /// <param name="mapVariable"></param>
+        /// <returns></returns>
+        public static string BuildParameterMap(this IEnumerable<ParameterGo> parameters, string mapVariable)
+        {
+            var builder = new StringBuilder();
+
+            builder.Append(mapVariable);
+            builder.Append(" := map[string]interface{} {");
+
+            if (parameters.Count() > 0)
+            {
+                builder.AppendLine();
+                var indented = new IndentedStringBuilder("  ");
+                parameters
+                    .Where(p => p.IsRequired)
+                    .OrderBy(p => p.SerializedName.ToString())
+                    .ForEach(p => indented.AppendLine("\"{0}\": {1},", p.NameForMap(), p.ValueForMap()));
+                builder.Append(indented);
+            }
+            builder.AppendLine("}");
+            return builder.ToString();
+        }
+
+        /// <summary>
+        /// Return list of parameters for specified location passed in an argument.
+        /// Refactor -> Probably CodeModeltransformer, but even with 5 references, the other mkethods are not used anywhere
+        /// </summary>
+        /// <param name="parameters"></param>
+        /// <param name="location"></param>
+        /// <returns></returns>
+        public static IEnumerable<ParameterGo> ByLocation(this IEnumerable<ParameterGo> parameters, ParameterLocation location)
+        {
+            return parameters
+                .Where(p => p.Location == location)
+                .AsEnumerable();
```

Comment:
I don't think this is necessary, wouldn't just passing back the result of the .Where() sufficient?